DataMigration/src/test/java/org/sagebionetoworks/dian/datamigration/SynapseTests.java

```diff
@@ -0,0 +1,307 @@
+import org.sagebionetworks.repo.model.EntityHeader;
+import org.sagebionetworks.repo.model.EntityType;
+import org.sagebionetworks.repo.model.FileEntity;
+import org.sagebionetworks.repo.model.Project;
+import org.sagebionetworks.repo.model.file.FileHandleAssociateType;
+import org.sagebionetworks.repo.model.file.FileHandleAssociation;
+import org.sagebionetworks.utils.MD5ChecksumHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class SynapseTests {
+
+    private static SynapseClient synapse;
+
+    private static Project project;
+    private static String projectId = "syn25791302";
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        // Create 2 users
+        synapse = new SynapseClientImpl();
+        synapse.setBearerAuthorizationToken(SynapseUtil.synapsePersonalAccessToken);
+
+        project = new Project();
+        project.setId(projectId);
+    }
+
+    @Test
+    public void testNavigateFolderStructure() throws SynapseException {
+        EntityChildrenRequest request = new EntityChildrenRequest();
+        request.setParentId(project.getId());
+        request.setIncludeTypes(Lists.newArrayList(EntityType.folder));
+        EntityChildrenResponse response = synapse.getEntityChildren(request);
+        assertNotNull(response);
+        assertNotNull(response.getPage());
+        assertEquals(2, response.getPage().size());
+
+        EntityHeader header = response.getPage().get(0);
+        assertNotNull(header);
+        assertEquals("2021-07-08", header.getName());
+
+        EntityHeader header2 = response.getPage().get(1);
+        assertNotNull(header2);
+        assertEquals("2021-09-21", header2.getName());
+
+        EntityChildrenRequest subFolderRequest = new EntityChildrenRequest();
+        subFolderRequest.setParentId(header.getId());
+        subFolderRequest.setIncludeTypes(Lists.newArrayList(EntityType.folder));
+        EntityChildrenResponse subFolderResponse = synapse.getEntityChildren(subFolderRequest);
+        assertNotNull(subFolderResponse);
+        assertNotNull(subFolderResponse.getPage());
+        assertEquals(3, subFolderResponse.getPage().size());
+
+        EntityHeader headerTestSession = subFolderResponse.getPage().get(0);
+        assertNotNull(header);
+        assertEquals("test_session", headerTestSession.getName());
+
+        EntityHeader headerTestSessionSchedule = subFolderResponse.getPage().get(1);
+        assertNotNull(header);
+        assertEquals("test_session_schedule", headerTestSessionSchedule.getName());
+
+        EntityHeader headerTestWakeSleep = subFolderResponse.getPage().get(2);
+        assertNotNull(header);
+        assertEquals("wake_sleep_schedule", headerTestWakeSleep.getName());
+    }
+
+    @Test
+    public void testDownload_testSession_ZIP() throws SynapseException, IOException {
```

Comment:
Not clear what testDownload_testSession_ZIP is testing. There's no reference to the logic in SynapseUtil and this code need not test that the Synapse client works.